From 744d0ea3ce3115ed6f1ae32bcb98511cdd41cf74 Mon Sep 17 00:00:00 2001
From: mickybart <mickybart@pygoscelis.org>
Date: Sat, 19 Nov 2016 19:05:05 -0500
Subject: [PATCH 51/52] OMS: StrictMode and files under /data/system/theme/

Themes are using /data/system/theme/ to push some files like LowBattery.ogg (audio notification)
When the device battery trigger the low battery state, the sound is not played due
to StrictMode and SystemUI is crashing.

So we need that StrictMode authorize files under /system OR /data/system/theme

Logcat of the issue:

E AndroidRuntime: Caused by: android.os.FileUriExposedException: file:///data/system/theme/audio/ui/LowBattery.ogg exposed beyond app through Notification.sound
E AndroidRuntime:        at android.os.StrictMode.onFileUriExposed(StrictMode.java:1799)
E AndroidRuntime:        at android.net.Uri.checkFileUriExposed(Uri.java:2346)
E AndroidRuntime:        at android.app.NotificationManager.notifyAsUser(NotificationManager.java:300)

Change-Id: I154dc4280de8eaf891772a9632283e9f547f5718
(cherry picked from commit 838f6466d39a100f9709ac253a6d7358ca66829f)
---
 core/java/android/net/Uri.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/java/android/net/Uri.java b/core/java/android/net/Uri.java
index 09af05c0dfb..3fd40428efb 100644
--- a/core/java/android/net/Uri.java
+++ b/core/java/android/net/Uri.java
@@ -2342,7 +2342,8 @@ public abstract class Uri implements Parcelable, Comparable<Uri> {
      * @hide
      */
     public void checkFileUriExposed(String location) {
-        if ("file".equals(getScheme()) && !getPath().startsWith("/system/")) {
+        if ("file".equals(getScheme()) && !(getPath().startsWith("/system/")
+                || getPath().startsWith("/data/system/theme/"))) {
             StrictMode.onFileUriExposed(this, location);
         }
     }
-- 
2.11.1

